tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -61,16 +131,160 @@ def startsWith(substring):
+        lambda value: value.isspace(),
+        lambda value: f'{value} is not blank.'
+    )
+
+def isInStringRange(lower, upper):
+    """Validate that string value is in a specific range."""
+    return make_validator(
+        lambda value: int(value) >= lower and int(value) <= upper,
+        lambda value: f'{value} is not in range [{lower}, {upper}].'
+    )
+
+def isStringLargerThan(val):
+    """Validate that string value is larger than val."""
+    return make_validator(
+        lambda value: int(value) > val,
+        lambda value: f'{value} is not larger than {val}.'
+    )
+
 def notEmpty(start=0, end=None):
     """Validate that string value isn't only blanks."""
     return make_validator(
-        lambda value: not value[start:end if end else len(value)].isspace(),
-        lambda value: f'{value} contains blanks between positions {start} and {end if end else len(value)}.'
+        lambda value: not str(value)[start:end if end else len(str(value))].isspace(),
+        lambda value: f'{str(value)} contains blanks between positions {start} and {end if end else len(str(value))}.'
+    )
+
+def isEmpty(start=0, end=None):
+    """Validate that string value is only blanks."""
+    return make_validator(
+        lambda value: value[start:end if end else len(value)].isspace(),
+        lambda value: f'{value} is not blank between positions {start} and {end if end else len(value)}.'
+    )
+
+
+def notZero(number_of_zeros=1):
+    """Validate that value is not zero."""
+    return make_validator(
+        lambda value: value != '0' * number_of_zeros,
+        lambda value: f'{value} is zero.'
+    )
+
+
+def isLargerThan(LowerBound):
+    """Validate that value is larger than the given value."""
+    return make_validator(
+        lambda value: float(value) > LowerBound if value is not None else False,
+        lambda value: f'{value} is not larger than {LowerBound}.'
+    )
+
+
+def isSmallerThan(UpperBound):
+    """Validate that value is smaller than the given value."""
+    return make_validator(
+        lambda value: value < UpperBound,
+        lambda value: f'{value} is not smaller than {UpperBound}.'
+    )
+
+def isLargerThanOrEqualTo(LowerBound):
+    """Validate that value is larger than the given value."""
+    return make_validator(
+        lambda value: value >= LowerBound,
+        lambda value: f'{value} is not larger than {LowerBound}.'
+    )
+
+def isSmallerThanOrEqualTo(UpperBound):
+    """Validate that value is smaller than the given value."""
+    return make_validator(
+        lambda value: value <= UpperBound,
+        lambda value: f'{value} is not smaller than {UpperBound}.'
+    )
+
+def isInLimits(LowerBound, UpperBound):
+    """Validate that value is in a range including the limits."""
+    return make_validator(
+        lambda value: value >= LowerBound and value <= UpperBound,
+        lambda value: f'{value} is not larger and equal to {LowerBound} and smaller and equal to {UpperBound}.'
```

Comment:
```suggestion
        lambda value: f'{value} is not larger or equal to {LowerBound} and/or smaller or equal to {UpperBound}.'
```

Reply:
Didi the change except kept 'and' in between instead of 'and/or'